wgpu-core/src/device/resource.rs

```diff
@@ -1276,6 +1276,19 @@ impl<A: HalApi> Device<A> {
                 .contains(wgt::DownlevelFlags::MULTISAMPLED_SHADING),
         );
 
+        let debug_source = if desc.debug {
+            Some(hal::DebugSource {
+                file_name: Cow::Owned(
+                    desc.label
+                        .as_ref()
+                        .map_or("shader".to_string(), |l| l.to_string()),
```

Comment:
a bit of a stretch to use this as a filename 🤔

Reply:
I can't determ wheather the origin source is really a wgsl or glsl, or we just use "shader.wgsl" for default?